### ISSUE TITLE
Improve Parquet read Performance

### ISSUE THREAD

**Issue by baibaichen** [ClickHouse/ClickHouse#19873]
This issue is related with https://github.com/ClickHouse/ClickHouse/issues/17623#issuecomment-770318813

It's been such a long time since i did this comparison, so I forgot some details.  I have  a parquet file about 500 columns, and all are int32 type，for fairness:

1. The parquet file is in the local disc.
2. I didn't create any index for merge tree

The query is a very simple aggregation query,   i.e.

``` SQL
select count(xx), sum (yy) from table where x1 = dd and y1 =xxx and ...
```

So, the performance depends on scan. I expect merge tree and parquet should have comparable performance，but querying parquet is quite slow and it looks like scanning all columns, and cache nothing.

What I am sure is encoding，I didn't use any encoding for merge tree, but parquet use dictionary encoding by default.  Here are some summary

|                                   | Merge tree | parquet                                                      |
| --------------------------------- | ---------- | ------------------------------------------------------------ |
| usage of indexes                  | no         | no                                                           |
| data compression                  | LZ4        | LZ4                                                          |
| ability to read subset of columns | yes        | not sure, but it looks like clickhouse doesn't prune columns |
| possible extra copying            | not sure   | not sure                                                     |
| inefficient data conversions      | No ecoding | dictionary encoding

**Comment by filimonov** [comment]
Can you also check memory usage? (see #10461)

**Comment by alanpaulkwan** [comment]
Could we use DuckDB's Parquet reader (from Impala)? If I recall correct from an email with Clickhouse-feedback circa 2019 you guys use parquetreader --> json but this has got to be inefficient?

**Comment by filimonov** [comment]
>  you guys use parquetreader --> json 

Sure we don't. We use https://github.com/apache/arrow/blob/master/cpp/src/parquet/arrow/reader.h

**Comment by alanpaulkwan** [comment]
Don't know what the problem is then!

**Comment by alexey-milovidov** [comment]
Fixed in #30936.
